### PR TITLE
Code syntax highlighting

### PR DESCRIPTION
**Pull request by Darxeal**
This adds code syntax highlighting using the [highlight.js](https://highlightjs.org/) library. It automatically detects the language. I added styles for both Light and Dark themes. Some examples:

image

image

I made it a part of the `Logged` class, so it works in pages like FAQ and exams:

image

### CODE REVIEW

**Comment by keombre** [comment]
Great work!

Just couple of small issues:
- Licence should be updated to include BSD 3-Clause License with reference to `hightlight.js`,
- I'm not sure if text from help boxes in task solution should be highlighted as well.
Perhaps use `highlightAuto` to determine the language relevance and set some reasonable cutoff threshold (see https://highlightjs.readthedocs.io/en/latest/api.html#highlightauto-value-languagesubset)

Other than that it looks like it's merge ready 👍

File: content/end.js [hunk 1]
```diff
@@ -121,6 +123,12 @@ class Logged {
         if (this.tButton && !this.tButton.getAttribute('style') && (this.oldScroll <= window.scrollY || !this.oldScroll))
             this.tButton.style.transform = "scale(1)"
     }
+
+    highlightCode() {
+        document.querySelectorAll('pre, code, tt').forEach((block) => {
+            hljs.highlightBlock(block);
```
Comment (keombre) [inline content/end.js:129]: We should format code only if language relevance is above a threshold.
For example:
```js
let formatted = hljs.highlightAuto(block.innerText)
if (formatted.r > 5) block.innerHTML = formatted.value
block.classList.add('hljs')
```

**Review by keombre — CHANGES_REQUESTED** [review]
Add a copy of `highlight.js` licence

**Comment by Darxeal** [comment]
I'm not a fan of the relevance solution. For example, it doesn't want to highlight code like `public int function()`, because this is valid in a lot of languages so the relevance is very low.

I suggest another solution - highlight.js lets you download a custom library for only the languages you want. So we could get one that supports only C++ and Java, because those are the only languages taught in progtest, as far as I know. There's also Bash, but that doesn't have very specific syntax rules so I would exclude it. 

That should mean that code blocks that don't support C++ or Java syntax will not get highlighted.

**Review by keombre — APPROVED** [review]

**Pull request merged**